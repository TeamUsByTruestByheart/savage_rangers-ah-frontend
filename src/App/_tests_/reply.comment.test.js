import React from 'react';
import { shallow } from '../../enzyme';
import ReplyComment from '../Components/Comment/reply.comment.body';
import actions from '../../Redux/Actions/getAllComments';

const { postCommentReply } = actions;

const props = {
  isHidden: true,
  replyCommentBody: postCommentReply,
  replies: [{
    id: 90,
    body: 'gdfhvjb',
    userId: 2,
    articleSlug: 'How-to-create-sequalize-seeds',
    parentCommentId: 83,
    isBlocked: false,
    iteration: 0,
    isEdited: false,
    createdAt: '2019-09-05T08:50:55.295Z',
    updatedAt: '2019-09-05T08:50:55.295Z'
  }]
};
const replyWrapper = shallow(<ReplyComment {...props} />);

describe('test the component', () => {
  test('should test the comment holder to render appropriate jsx', () => {
    expect(replyWrapper.find('div').exists()).toBe(true);
    expect(replyWrapper).toMatchSnapshot();
  });

  test('should simulate onChange Textarea', () => {
    replyWrapper.find('textarea').simulate('change', { target: { name: 'password', value: 'this is your comment' } });
  });

  test('should simulate onClick Button', () => {
    replyWrapper.find('textarea').simulate('click', { target: { name: 'password', value: 'this is your comment' } });
  });

  test('should simulate the reply button click', () => {
    const mockAction = jest.fn();
    replyWrapper.find('button').simulate('click');
    expect(mockAction.mock.calls.length).toEqual(0);
  });
});
