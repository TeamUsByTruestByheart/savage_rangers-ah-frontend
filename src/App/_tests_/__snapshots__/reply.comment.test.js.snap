// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the component should test the comment holder to render appropriate jsx 1`] = `ShallowWrapper {}`;
